File: cpp/tests/MajorityElementTest.cpp
#include "MajorityElement.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MajorityElement, Solution169_1) {
  std::vector<int> Nums1{3, 2, 3};
  const int Expected1 = 3;
  EXPECT_EQ(Expected1, Solution169_1().majorityElement(Nums1));

  std::vector<int> Nums2{2, 2, 1, 1, 1, 2, 2};
  const int Expected2 = 2;
  EXPECT_EQ(Expected2, Solution169_1().majorityElement(Nums2));
}

File: java/src/test/java/io/binac/leetcode/MajorityElementTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MajorityElementTest {
    private static final MajorityElement.Solution1 SOLUTION1 = new MajorityElement.Solution1();

    private final int[] NUMS1 = {3, 2, 3};
    private final int EXPECTED1 = 3;

    private final int[] NUMS2 = {2, 2, 1, 1, 1, 2, 2};
    private final int EXPECTED2 = 2;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.majorityElement(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.majorityElement(NUMS2));
    }
}